src/xnnpack/simd/s16-avx2.h

```diff
@@ -18,16 +18,23 @@
 // SIMD vector type for s16 using AVX2.
 typedef __m256i xnn_simd_s16_t;
 #define xnn_simd_size_s16 16
-#define xnn_simd_log2_size_s16 3
+#define xnn_simd_log2_size_s16 4
 #define xnn_simd_bytes_s16 (xnn_simd_size_s16 * sizeof(int16_t))
 
 #define XNN_SIMD_CONST_S16(var, val) \
   const xnn_simd_s16_t var = _mm256_set1_epi16(val);
 
 // Mask table used for masked load/store operations.
-static const int32_t mask_table_avx_s32[14] = {-1, -1, -1, -1, -1, -1, -1,
-                                               0,  0,  0,  0,  0,  0,  0};
+static const int32_t mask_table_avx_s16[32] = {-1, -1, -1, -1, -1, -1, -1, -1,-1, -1, -1, -1, -1, -1, -1, -1,
+                                               0,  0,  0,  0,  0,  0,  0, 0,0,  0,  0,  0,  0,  0,  0, 0};
 // Arithmetic operations.
+static XNN_INLINE xnn_simd_s16_t xnn_low_cvt_s16_s32(xnn_simd_s16_t a) {
+  return _mm256_cvtepi16_epi32(_mm256_extracti128_si256(a,0));
```

Comment:
use _mm256_castsi256_si128 instead of extract as it's free

Reply:
Only for lower-bit extraction?

Reply:
Yes only to extract the lower bits.

_mm256_cvtepi16_epi32(_mm256_castsi256_si128(a)); should work. (Not tested)

Reply:
Will check and update